lib/python/treadmill/sproc/garbage_collector.py

```diff
@@ -0,0 +1,57 @@
+"""Garbage collector for LDAP and IPA.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import time
+import click
+from treadmill import plugin_manager
+from treadmill import cli
+
+_LOGGER = logging.getLogger(__name__)
+_MODULE = 'treadmill.garbage_collector'
+
+
+def _run_gc(gc_plugins, once, interval):
+    """Garbage collector plugins executor.
+    """
+    while True:
+        for name in gc_plugins or plugin_manager.names(_MODULE):
+            try:
+                plugin = plugin_manager.load(_MODULE, name)
+                plugin()
+            except Exception:  # pylint: disable=W0703
+                _LOGGER.exception('Error processing plugin: %s', name)
+
+        if once:
+            return
+        time.sleep(interval)
+
+
+def init():
+    """Return top level command handler.
+    """
+
+    @click.command()
+    @click.option('--interval', type=float, default=3600,
+                  help='interval in seconds')
+    @click.option('--once', is_flag=True, default=False, help='Run once.')
+    @click.option('--list-plugins', is_flag=True,
+                  help='List of available plugins.')
+    @click.option('--gc-plugins', default=None, type=cli.LIST,
+                  help='Comma separated list of plugins to run.')
+    def top(interval, once, list_plugins, gc_plugins):
+
+        """Garbage Collector.
+        """
+        if list_plugins:
+            print(", ".join(plugin_manager.names(_MODULE)))
+            exit()
+        _LOGGER.info('Running garbage collector')
+        _run_gc(gc_plugins, once, interval)
+        _LOGGER.info('Cleanup completed.')
```

Comment:
it will never be printed, right? unless it is --once... 

while True:
   _run_gc(gs_plugins)
   if once:
       _LOGGER.info('Done.')
       break
   time.sleep(interval)